kernel_gateway/services/kernels/manager.py

```diff
@@ -76,7 +76,16 @@ def start_kernel(self, *args, **kwargs):
                 client = kernel.client()
                 # Only start channels and wait for ready in HTTP mode
                 client.start_channels()
-                client.wait_for_ready()
+                try:
+                    client.wait_for_ready()
```

Comment:
This code only runs when there's a `seed_source` notebook configured. What if there isn't (which is the far more typical case)? The parent class `start_kernel` call is what actually spawns the kernel. Is kernel failure with the address error synchronous with that call and we can catch it there? Or not, and then we've got the problem of the client dying asynchronously at any point in the code base unless we wait for it before returning from `start_kernel` which is a new behavior.